### PR TITLE
Add quarkus:test goal

### PR DESCRIPTION
**Pull request by stuartwdouglas**
This enables continuous testing without
starting the application in dev mode.

### CODE REVIEW

File: devtools/gradle/src/main/java/io/quarkus/gradle/QuarkusPlugin.java [hunk 1]
```diff
@@ -163,6 +166,7 @@ public void execute(Task test) {
                     Task testResourcesTask = tasks.getByName(JavaPlugin.PROCESS_TEST_RESOURCES_TASK_NAME);
                     quarkusDev.dependsOn(classesTask, resourcesTask, testClassesTask, testResourcesTask, quarkusGenerateCode);
                     quarkusRemoteDev.dependsOn(classesTask, resourcesTask);
+                    quarkusTest.dependsOn(classesTask, resourcesTask, testClassesTask, testResourcesTask, quarkusGenerateCode);
```
Comment (Ladicek) [inline devtools/gradle/src/main/java/io/quarkus/gradle/QuarkusPlugin.java:169]: I'm not entirely sure, but maybe also `quarkusGenerateCodeTests`?

(Which should maybe be there for `quarkusDev` as well? As I said, not sure.)

Comment (Ladicek) [reply devtools/gradle/src/main/java/io/quarkus/gradle/QuarkusPlugin.java:169]: Maybe also CC @michalszynkiewicz

Comment (michalszynkiewicz) [reply devtools/gradle/src/main/java/io/quarkus/gradle/QuarkusPlugin.java:169]: 👍🏻 I think test code generation should also be among the dependencies

**Comment by stuartwdouglas** [comment]
@gsmet it would be great if we could get this into the next Alpha

**Review by gsmet — APPROVED** [review]

**Pull request merged**